blog/_posts/2025-08-14-node-autoprovisioning.md

```diff
@@ -0,0 +1,76 @@
+
+![nap-ga-announcement](/assets/images/aks-nap/nap-ga-announcement.jpg)
+
+## What’s New with NAP since Preview Launch
+
+- More networking options: Azure CNI with or without Overlay is supported, Cilium dataplane is supported; BYO CNI is allowed. Custom Virtual Networks (VNet) is supported. For more on networking, visit our [NAP networking documentation](https://learn.microsoft.com/azure/aks/node-autoprovision-networking)
+- New method of node bootstrapping; better reliability, from compatibility with AKS VM image releases and AKS bootstrapping configuration updates.
+- Improved upgrade experience, including support for [AKS Maintenance Windows](https://learn.microsoft.com/azure/aks/planned-maintenance) and [Karpenter disruption budgets](https://learn.microsoft.com/azure/aks/node-autoprovision-disruption#disruption-budgets), which allow users to control the speed of disruption in the cluster.
+- New Karpenter core capabilities integrated and supported: v1 NodePool API, expanded support for [disruption budgets](https://learn.microsoft.com/en-us/azure/aks/node-autoprovision-disruption) features such as terminationGracePeriod, consolidateAfter, forceful expiration, node repair, and more. [NodePool configuration files](https://learn.microsoft.com/azure/aks/node-autoprovision-node-pools) now have status conditions that indicate if they are ready. See core releases for details.
+- New Azure Karpenter Provider capabilities: support for Azure Linux (including v3), ephemeral disk placement, Linux admin username, custom kubelet configuration, tagging of Azure resources, artifact streaming, non-zonal regions and VM SKUs, zone constraint in NodePool requirements (and generally cleaner set of selectors), node auto-repair, network interface garbage collection, support for NVMe-only VM SKUs, AKS/Kubernetes 1.30-33, readiness status in AKSNodeClass, and more. See [Azure Karpenter provider releases](https://github.com/Azure/karpenter-provider-azure/releases) for details.
+- Improved observability, via rich set of metrics (accessible through [Managed Prometheus](https://learn.microsoft.com/azure/azure-monitor/metrics/prometheus-metrics-overview)), accessible and improved logs through [Azure Monitor](https://learn.microsoft.com/azure/azure-monitor/metrics/data-platform-metrics), NodePool / AKSNodeClass / NodeClaim conditions (including reasons for drift or provisioning failures), and events.
+- Default NAP node pools are now optional; [disabling NAP is supported](https://learn.microsoft.com/azure/aks/node-autoprovision#disabling-node-autoprovisioning).
+- Better performance, reliability, error handling (including handling of provisioning failures), and security. Numerous bugfixes and issues resolved.
+- [Extensive test coverage](https://github.com/Azure/karpenter-provider-azure/tree/main/test). Eight new E2E test suites, ~100 total scenarios. 95% unit test coverage.
+- Github Contributions welcome! (GitHub Codespace-based dev/test environment in 5 min)
+
```

Comment:
Could we have a section on how the Preview metrics / adoption has been so far on this.